### DISCUSSION TITLE
v2.1.0

### DISCUSSION THREAD

**Discussion by iSoron** [iSoron/uhabits#1501] [Announcements]
### Added
- Allow user to add notes to specific dates (@vbh, #1103)
- Allow user to track "at most" numerical habits (@KristianTashkov, #1101)
- Allow user to add skips to measurable habits (@kalina559, #1319)
- Bring back custom frequencies (x times in y days) (@hiqua, #1079)
- Improve number picker (@hiqua, @iSoron, #1082, #1370)
- Add new checkmark and number picker (@iSoron, #1370)
- Allow user to import numerical habits from HabitBull (@hiqua, #1278)
- Add support for Android 13 themed icons (@cheeeeer, #1497)

### Removed
- Hide snooze button on Android 12 notifications (@hiqua, #1226)
- Remove preference to set LED lights (@iSoron)

### Changed
- Hide failed habits along with completed ones (@hiqua, #1052)
- Cycle through all checkmark states when toggling (@iSoron)
- Add delay after toggling a habit (@hiqua, @kalina559, #1147)
- Small theme improvements (@KristianTashkov, #1113)
- Left-align habit notes (@iSoron)
- Increase target SDK to 31 (@hiqua)

### Fixed
- Fix small dialog buttons (@kalina559, #1096)
- Fix invalid CSV files (@hiqua, #1177)
- Fix small issues in calendar chart (@kalina559, #1314)
- Resort habit list after edit (@hiqua, #1350)
- Fix marker scaling in frequency display (@eduebernal, #1425)

### Refactoring & Testing
- Replace raster icons by vector assets (@kalina559)
- Remove JVM dependencies from uhabits-core module (@sgallese)
- Add various missing tests (@sgallese)
- Upgrade project dependencies (@hiqua, @sgallese)

<hr /><em>This discussion was created from the release <a href='https://github.com/iSoron/uhabits/releases/tag/v2.1.0'>v2.1.0</a>.</em>

**Comment by t214c** [comment]
Thanks for the update. Please see my comment on #1226 about keeping the snooze button.

**Comment by hiqua** [comment]
Awesome, thanks a lot @iSoron!

**Comment by Henwij** [comment]
After a long wait, I finally got to see this version. Awesome, thanks you @iSoron and all the contributors!

**Comment by woj-tek** [comment]
Issue links are not available on the release page (https://github.com/iSoron/uhabits/releases/tag/v2.1.0) though they are here and e-mail notification :-)

**Reply by iSoron** [reply]
I noticed; not sure why. It seems like a GitHub issue. The raw markdown is the same on both pages (in fact, this one was created automatically from that one).

**Comment by shanbhagrohan** [comment]
Facing Issues with the Tasker plugin since the latest update. It's visible in the plugin list but I can't setup the task
![Screenshot_20220913-080012_Tasker](https://user-images.githubusercontent.com/39913302/189794267-9adbf9d4-43e5-411d-bf76-e8011f16a9fd.jpg)

**Reply by StanczakDominik** [reply]
Yup, getting the same error here. Going to repost this as an issue...

#1503 created.

**Reply by shanbhagrohan** [reply]
My tasker version is 6.14 beta but its not a Tasker issue because it successfully executed a task yesterday before I updated loop.

Thank you for raising the issue.

**Comment by Picorims** [comment]
Thanks a lot for the update! Will it be deployed to F-Droid as well, or do I have to uninstall the existing version from my Android phone?

**Reply by JVariance** [reply]
👀

**Comment by nottheend** [comment]
Is there a timeline when this version gets into stabl Looking eagerly forward to the notes feature. Thanks for the great work :-)
